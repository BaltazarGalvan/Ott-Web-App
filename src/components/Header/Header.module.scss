@use '../../styles/variables';
@use '../../styles/theme';
@use '../../styles/mixins/responsive';
@use '../../styles/mixins/utils';

//
// Header
// --------------------------------

.header {
  height: variables.$header-height;
  padding: 10px variables.$base-spacing * 4;
  background: var(--header-background);

  //
  // Make header fixed
  //
  &.fixed {
    position: fixed;
    z-index: variables.$header-z-index;
    width: 100%;
  }

  //
  // Make header static
  //
  &.static {
    position: static;
    width: 100%;
  }
}

//
// Header container
//
.container {
  position: relative;
  display: flex;
  flex-direction: row;
  height: 100%;
}

//
// Header menu
//
.menu {
  display: none;
}

//
// Header navigation
//
.nav {
  display: inline-block;
  flex: 1;

  > a {
    margin: 0 6px;
  }
}

.search {
  width: 180px;
}

//
// mediaQueries
// --------------------------------

@include responsive.mobile-and-tablet() {
  .header {
    height: variables.$header-height-mobile;
    padding: 10px variables.$base-spacing * 2;
  }

  .menu {
    display: block;
  }

  .nav {
    display: none;
  }
}

@include responsive.mobile-only() {
  .header {
    padding: 10px variables.$base-spacing;
  }

  .menu {
    transition: opacity 0.2s ease;
  }

  .search {
    width: 25px;
  }
}

@include responsive.tablet-only() {
  .menu {
    width: 180px;
  }
}

@include responsive.desktop-only() {
  .nav {
    text-align: center;
  }
}
