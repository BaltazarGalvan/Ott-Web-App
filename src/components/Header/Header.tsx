import React, { ReactFragment } from 'react';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';

import SearchBar, { Props as SearchBarProps } from '../SearchBar/SearchBar';
import Logo from '../Logo/Logo';
import Menu from '../../icons/Menu';
import SearchIcon from '../../icons/Search';
import CloseIcon from '../../icons/Close';
import IconButton from '../../components/IconButton/IconButton';
import useBreakpoint, { Breakpoint } from '../../hooks/useBreakpoint';

import styles from './Header.module.scss';

type TypeHeader = 'static' | 'fixed';

type Props = {
  headerType?: TypeHeader;
  onMenuButtonClick: () => void;
  logoSrc?: string;
  searchBarProps: SearchBarProps;
  searchEnabled: boolean;
  searchActive: boolean;
  onSearchButtonClick?: () => void;
  onCloseSearchButtonClick?: () => void;
  children?: ReactFragment;
};

const Header: React.FC<Props> = ({
  children,
  headerType = 'static',
  onMenuButtonClick,
  logoSrc,
  searchBarProps,
  searchActive,
  onSearchButtonClick,
  searchEnabled,
  onCloseSearchButtonClick,
}) => {
  const { t } = useTranslation('menu');
  const breakpoint = useBreakpoint();
  const headerClassName = classNames(styles.header, styles[headerType], {
    [styles.mobileSearchActive]: searchActive && breakpoint <= Breakpoint.sm,
  });

  const search =
    breakpoint <= Breakpoint.sm ? (
      searchActive ? (
        <div className={styles.mobileSearch}>
          <SearchBar {...searchBarProps} />
          <IconButton
            className={styles.iconButton}
            aria-label="Close search"
            onClick={() => {
              if (onCloseSearchButtonClick) {
                onCloseSearchButtonClick();
              }
            }}
          >
            <CloseIcon />
          </IconButton>
        </div>
      ) : (
        <IconButton
          className={styles.iconButton}
          aria-label="Open search"
          onClick={() => {
            if (onSearchButtonClick) {
              onSearchButtonClick();
            }
          }}
        >
          <SearchIcon />
        </IconButton>
      )
    ) : (
      <SearchBar {...searchBarProps} />
    );

  return (
    <header className={headerClassName}>
      <div className={styles.container}>
        <div className={styles.menu}>
          <IconButton className={styles.iconButton} aria-label={t('open_menu')} onClick={onMenuButtonClick}>
            <Menu />
          </IconButton>
        </div>
        {logoSrc && (
          <div className={styles.brand}>
            <Logo src={logoSrc} />
          </div>
        )}
        <nav className={styles.nav} aria-label="menu">
          {children}
        </nav>
        <div className={styles.search}>{searchEnabled ? search : null}</div>
      </div>
    </header>
  );
};
export default Header;
