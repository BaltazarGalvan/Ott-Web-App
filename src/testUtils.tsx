import { BrowserRouter as Router } from 'react-router-dom';
import React, { ReactNode, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

interface WrapperProps {
  children?: ReactNode;
}

export const wrapper = ({ children }: WrapperProps) => <Router>{children}</Router>;

const customRender = (ui: ReactElement, options?: RenderOptions) => render(ui, { wrapper, ...options });

export const mockWindowLocation = (path: string) => {
  Object.defineProperty(window, 'location', {
    value: {
      pathname: path,
      assign: jest.fn(),
    },
  });
};

export const mockMatchMedia = () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    })),
  });
};

export { customRender as render };
