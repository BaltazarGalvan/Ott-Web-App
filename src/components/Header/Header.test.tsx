import React from 'react';

import { render } from '../../testUtils';

import Header from './Header';

describe('<Header />', () => {
  test('renders headers', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
