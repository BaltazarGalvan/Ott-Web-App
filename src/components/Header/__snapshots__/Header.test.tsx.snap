// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders header 1`] = `
<div>
  <header
    class="header static"
  >
    <div
      class="container"
    >
      <div
        class="menu"
      >
        <div
          aria-label="open menu"
          class="iconButton"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="icon"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </div>
      </div>
      <nav
        aria-label="menu"
        class="nav"
      >
        <a
          aria-current="page"
          class="button default outlined active"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
      </nav>
      <div
        class="search"
      />
    </div>
  </header>
</div>
`;
